DSharpPlus/Entities/Guild/DiscordRole.cs

```diff
@@ -135,8 +155,8 @@ public Task ModifyPositionAsync(int position, string reason = null)
         /// <exception cref="Exceptions.NotFoundException">Thrown when the role does not exist.</exception>
         /// <exception cref="Exceptions.BadRequestException">Thrown when an invalid parameter was provided.</exception>
         /// <exception cref="Exceptions.ServerErrorException">Thrown when Discord is unable to process the request.</exception>
-        public Task ModifyAsync(string name = null, Permissions? permissions = null, DiscordColor? color = null, bool? hoist = null, bool? mentionable = null, string reason = null)
-            => this.Discord.ApiClient.ModifyGuildRoleAsync(this._guild_id, this.Id, name, permissions, color?.Value, hoist, mentionable, reason);
+        public Task ModifyAsync(string name = null, Permissions? permissions = null, DiscordColor? color = null, bool? hoist = null, bool? mentionable = null, string reason = null, Stream icon = null, DiscordEmoji emoji = null)
+            => this.Discord.ApiClient.ModifyGuildRoleAsync(this._guild_id, this.Id, name, permissions, color?.Value, hoist, mentionable, reason, icon, emoji);
```

Comment:
what's with ModifyAsync(Action<RoleEditModel> action)

Reply:
My commits are never to be taken at face-value. They tend to be incomplete and sporadic.

Reply:
I'm currently bouncing between six different files, and I'm aware of what I missed.

Reply:
Actually wait, what? What're you referring to here

Reply:
I would set reason to the last position